src/factories/meta/CollectionCollector_factory.h

```diff
@@ -35,6 +35,9 @@ namespace eicrecon {
         }
         typename T::collection_type* merged_collection = m_output().get();
         m_algo->process(in_collections, merged_collection);
+        //Log how many hits were collected from N input collections
+        this->logger()->debug("Collected {} hits from {} input collections", merged_collection->size(), in_collections.size());
+
```

Comment:
These need to be in the algorithm. Factories are supposed to not contain custom logic.

Reply:
Done